src/sagemaker/dataset_definition/inputs.py

```diff
@@ -0,0 +1,256 @@
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""The input configs for DatasetDefinition.
+
+DatasetDefinition support the data sources like S3 data queries via Athena
+and Redshift. A mechanism has to be created for customers to generate datasets
+from Athena/Redshift queries and to retrieve the data, using Processing jobs
+so as to make it available for other downstream processes.
+"""
+from __future__ import absolute_import
+
+from typing import Dict, Any
+import attr
+
+_service_attribute_name = "service_attribute_name"
+
+
+@attr.s
+class _BaseConfig:
+    """Base config object for DatasetDefinition.
+
+    The class implemented common to_dict() and from_dict() methods to
+    serialize/deserialize the class to construct service request.
+    """
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Construct the dictionary from the class.
+
+        Returns:
+            dict represents the class.
+        """
+
+        dictionary = {}
+
+        for attribute in self.__class__.get_attributes():
+            if self.__dict__[attribute.name] is not None:
+                attribute_value = self.__dict__[attribute.name]
+                if isinstance(self.__dict__[attribute.name], _BaseConfig):
+                    dictionary[
+                        attribute.metadata[_service_attribute_name]
+                    ] = attribute_value.to_dict()
+                else:
+                    dictionary[attribute.metadata[_service_attribute_name]] = attribute_value
+
+        return dictionary
+
+    @classmethod
+    def from_dict(cls, service_response) -> "_BaseConfig":
+        """Construct the _BaseConfig object from the dictionary.
+
+        args:
+            service_response: the json response returned from the service.
+
+         Returns:
+             a _BaseConfig object.
+        """
+
+        if service_response is None:
+            return None
+
+        dd_dict = {}
+
+        for attribute in attr.fields(cls):
+            service_attr_name = attribute.metadata[_service_attribute_name]
+            if service_attr_name in service_response:
+                if isinstance(service_response[service_attr_name], dict):
+                    dd_dict[attribute.name] = attribute.type.from_dict(
+                        service_response[service_attr_name]
+                    )
+                else:
+                    dd_dict[attribute.name] = service_response[
+                        attribute.metadata[_service_attribute_name]
+                    ]
+
+        return cls(**dd_dict)
+
+    @classmethod
+    def get_attributes(cls):
+        """Get all class attributes
+
+        Returns:
+            dict represents the class.
+        """
+        return attr.fields(cls)
+
+
+@attr.s
+class RedshiftDatasetDefinition(_BaseConfig):
```

Comment:
I thought we have decided to use dict for these. Did i remember it wrong?

Reply:
So I think we decided not to use `Config` defined in sagemaker.feature_store to avoid coupling. 

_BaseConfig is created for DatasetDefinition. 

IMO this is better than dictionary, we don't need extra logic for each class to perform serialization and deserialization.

Reply:
if you use a `dict`, you can do just like you did on line 85 and eliminate all the machinery (plus, that's more in the spirit of `attrs`, which you're using in an attempt to get rid of boilerplate)

Reply:
Thanks - send a new revision to use ApiObject.